lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -598,7 +595,7 @@ def set_password(self, trans, id, payload={}, **kwd):
         """
         user, message = self.user_manager.change_password(trans, id=id, **payload)
         if user is None:
-            raise MessageException(message)
+            raise exceptions.AuthenticationRequired(message)
```

Comment:
```suggestion
            raise exceptions.AuthenticationFailed(message)
```

Reply:
I specifically want 403 here, 401 implies you can authenticate using headers I think

Reply:
Uh, you are right, 401 should be used only if the client needs to authenticate using HTTP Authentication, which we don't use. Should we remove this class and corresponding error code to prevent misuse?

Reply:
Maybe we could just change it to 403? But I would do it in a different PR since that will probably break a few things. E.g. some of our client javascript checks specific error codes.

Reply:
Filed an issue: https://github.com/galaxyproject/galaxy/issues/7407